conans/client/graph/graph_builder.py

```diff
@@ -269,15 +271,23 @@ def _resolve_replace_requires(self, node, require, profile_build, profile_host,
             graph.replaced_requires[original_require] = repr(require.ref)
             break  # First match executes the alternative and finishes checking others
 
+    def _match_host_version_ref_track(self, require):
+        ref_track_re = re.compile("<host_version(?::(.+))?>")
+        return ref_track_re.match(str(require.ref.version))
+
     def _create_new_node(self, node, require, graph, profile_host, profile_build, graph_lock):
-        if require.ref.version == "<host_version>":
+        ref_track_match = self._match_host_version_ref_track(require)
+        if require.ref.version == "<host_version>" or ref_track_match:
```

Comment:
Change it for a faster ``if "<host_version>" in require.ref.version`` or something in that line, and only when that is true, start parsing/regex or processing the specific syntax.  Probably just a ``split(":")`` after that is good enough too.